File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/mediator/mediator/Colleague.java
package com.bigboxcode.designpattern.mediator.mediator;

public abstract class Colleague {
    protected Mediator mediator;

    public Colleague(Mediator mediator) {
        this.mediator = mediator;
    }

    public abstract void sendMessage(Colleague colleague, String message);

    public abstract void receiveMessage(String message);
}

File: design-patterns/ts/mediator/general/colleague.ts
// colleague.ts

import Mediator from "./mediator";

abstract class Colleague {
    protected mediator: Mediator;
    constructor(mediator: Mediator) {
        this.mediator = mediator;
    }

    abstract sendMessage(colleague: Colleague, message: string): void;
    abstract receiveMessage(message: string): void;
}

export default Colleague;